lib/chef/knife/bootstrap.rb

```diff
@@ -370,14 +367,12 @@ class Bootstrap < Knife
                                         ],
       }.freeze
 
-      DEPRECATED_FLAGS.each do |flag, new_flag_config|
-        new_flag, old_long = new_flag_config
-        new_long = options[new_flag][:long]
-        new_flag_name = new_long.split(" ").first
-
-        option(flag, long: new_long,
-               description: "#{old_long} is deprecated. Use #{new_long} instead.",
-               boolean: options[new_flag][:boolean])
+      DEPRECATED_FLAGS.each do |deprecated_key, deprecation_entry|
+        new_key, deprecated_long = deprecation_entry
+        new_long = options[new_key][:long]
```

Comment:
Outside of rename/relocate in file, the actual change was here. We were previously specifying `long: new_long`, which was incorrect. This now correctly uses `deprecated_long`.